query/src/servers/http/clickhouse_handler.rs

```diff
@@ -207,15 +303,46 @@ pub async fn clickhouse_handler_post(
         .map_err(BadRequest)?;
 
     let mut sql = params.query();
-    sql.push_str(body.into_string().await?.as_str());
-    let (plan, format) = PlanParser::parse_with_format(ctx.clone(), &sql)
-        .await
-        .map_err(BadRequest)?;
 
-    ctx.attach_query_str(&sql);
-    execute(ctx, plan, format, None, params)
-        .await
-        .map_err(InternalServerError)
+    let (stmts, _) = DfParser::parse_sql(sql.as_str(), ctx.get_current_session().get_type())
+        .unwrap_or_else(|_| (vec![], vec![]));
+
+    let settings = ctx.get_settings();
+    if settings.get_enable_new_processor_framework().unwrap() != 0
+        && !ctx.get_config().query.management_mode
+        && ctx.get_cluster().is_empty()
+        && settings.get_enable_planner_v2().unwrap() != 0
```

Comment:
```suggestion
        && settings.get_enable_planner_v2()? != 0
```

Reply:
Because the function returns `PoemResult<impl IntoResponse>`, we should add `map_err()?` to replace `?` 

It seems ok because `unwrap()` doesn't failed here?

Reply:
I think it's better not to use a `unwrap`, since it may fail one day, and we won't know that until the process panicked...

Reply:
Ok.

Reply:
Maybe it's better to implement `Into<PoemError> for ErrorCode`?

Reply:
> Maybe it's better to implement `Into<PoemError> for ErrorCode`? (not related to this PR)

Created an issue for it: https://github.com/datafuselabs/databend/issues/6137